app/categorization/llm_categorization.py

```diff
@@ -1,169 +1,71 @@
-from datetime import datetime
+from typing import Dict, Optional, Union
 import json
-from typing import Dict, Any, Union
 from langchain_community.chat_models import ChatOllama
-from langchain_core.prompts import PromptTemplate
+from promptTemplate import(
+prompt,
+Aprompt,
+Dprompt)
+from llm_helper import (
+SexLevel, 
+AgeFormat)
 
 
-def SexLevel(result_dict: Dict[str, str], r: str, key: str) -> Dict[str, Any]:
-    value = result_dict[key]
-    value = value.lower()
-    var1: str = ""
-    var2: str = ""
-    var3: str = ""
-
-    if "1" in value or "2" in value or "3" in value:
-        var1 = "1"
-        var2 = "2"
-        var3 = "3"
-    elif "0" in value or "1" in value or "2" in value:
-        var1 = "0"
-        var2 = "1"
-        var3 = "2"
-    elif "m" in value or "f" in value or "o" in value:
-        var1 = "m"
-        var2 = "f"
-        var3 = "o"
-    elif "male" in value or "female" in value or "other" in value:
-        var1 = "male"
-        var2 = "female"
-        var3 = "other"
-    else:
-        output: Dict[str, Union[str, Dict[str, str]]] = (
-            {}
-        )  # Or any default output as per your requirement
+def D(key: str, value: str) -> Optional[Dict[str, str]]:
+    llm = ChatOllama(model="gemma")
+    chainD = Dprompt | llm
+    llm_response2 = chainD.invoke({"column": key, "content": value})
```

Comment:
I think it makes sense to use informative names - I am not entirely sure what `llm_response2` does (probably it is the second run through the llm??). Is there a more suitable name?

Reply:
Yes its the one that checks for diagnosis here 
and below its the one that checks for assesment as those were different functions having the same names didnt affect the working I will change those to something informative